guacamole/src/main/frontend/src/app/client/types/ManagedFilesystem.js

```diff
@@ -335,6 +348,72 @@ angular.module('client').factory('ManagedFilesystem', ['$rootScope', '$injector'
 
     };
 
+    /**
+     * translate the permission number to the permission string
+     *
+     * @param {Number} permission
+     */
+    ManagedFilesystem.permissionTranslate = function permissionTranslate(permission) {
+
+        var Ow_R = 256;
+        var Ow_W = 128;
+        var Ow_X = 64;
+        var Gp_R = 32;
+        var Gp_W = 16;
+        var Gp_X = 8;
+        var Ot_R = 4;
+        var Ot_W = 2;
+        var Ot_X = 1;
+
+        let res = '';
+        if ((permission & Ow_R) === Ow_R) {
+            res += 'r';
+        } else {
+            res += '-';
+        }
+        if ((permission & Ow_W) === Ow_W) {
+            res += 'w';
+        } else {
+            res += '-';
+        }
+        if ((permission & Ow_X) === Ow_X) {
+            res += 'x';
+        } else {
+            res += '-';
+        }
+        if ((permission & Gp_R) === Gp_R) {
+            res += 'r';
+        } else {
+            res += '-';
+        }
+        if ((permission & Gp_W) === Gp_W) {
+            res += 'w';
+        } else {
+            res += '-';
+        }
+        if ((permission & Gp_X) === Gp_X) {
+            res += 'x';
+        } else {
+            res += '-';
+        }
+        if ((permission & Ot_R) === Ot_R) {
+            res += 'r';
+        } else {
+            res += '-';
+        }
+        if ((permission & Ot_W) === Ot_W) {
+            res += 'w';
+        } else {
+            res += '-';
+        }
+        if ((permission & Ot_X) === Ot_X) {
+            res += 'x';
+        } else {
+            res += '-';
+        }
```

Comment:
There's got to be a more readable way to do this. Perhaps you can structure things as a loop and avoid this kind of brute-force list?